### PR TITLE
Update the system model to focus more on the core entities.

### PR DESCRIPTION
**Pull request by mbruggmann**
Now that we have a way of labeling and annotating catalogue
entities, we can focus the system model more on the entities
themselves. I believe that also makes more sense as a model that
is shared across many companies - it's easier to standardize on
the core entities, and leave some room to implement further
abstraction or categories to individual implementations.

### CODE REVIEW

File: docs/features/software-catalog/system-model.md [hunk 1]
```diff
@@ -3,55 +3,24 @@ id: system-model
...
 the concepts are not yet supported in Backstage._
 
-## Concepts
+## Core Entities
 
-We model our technology using these five concepts (further explained below):
+We model software in the Backstage catalogue using these three core entities
+(further explained below):
 
-- **Domains** are a high-level grouping of systems
```
Comment (Fox32) [inline docs/features/software-catalog/system-model.md:17]: Till now I expected Domains and Systems to be entities, too? Does that change, or will it be added again once they are implemented?

**Comment by mbruggmann** [comment]
@stefanalund Also switched to a white background on the diagram (rather than transparent) so it should render better on the dark microsite.

File: docs/features/software-catalog/system-model.md (hunk 1 above)
Comment (mbruggmann) [reply docs/features/software-catalog/system-model.md:17]: We had a lot of internal discussion around not enforcing the system concept for all components. Shared libraries or modules in a mobile application might not have a system to attach them to (or sometimes, several ones). It'd still be nice if Backstage would have some support for systems though, even if associating entities with a system is optional. If you think that is valuable, we can keep it around?

Comment (Fox32) [reply docs/features/software-catalog/system-model.md:17]: Till now they looked quite optional, but in bigger platform I would expect them to be essential. 

I don't think that having the concept would enforce it for the users. I know Systems and Domain don't exist yet, but it is something we would be happy to contribute once the basics of #1964 are done.

Comment (mbruggmann) [reply docs/features/software-catalog/system-model.md:17]: Alright, I added them back. What do you think?

Comment (Fox32) [reply docs/features/software-catalog/system-model.md:17]: Looks good, later when they are implemented in backstage we can add another picture to `Ecosystem Modeling` how these are part of the model.

Comment (mbruggmann) [reply docs/features/software-catalog/system-model.md:17]: Yeah, for now I wanted to leave us some room to figure out how exactly to do it.

**Review by stefanalund — APPROVED** [review]
Thanks -- @freben may want to have a look before we merge.

(FYI I also am in favour of keeping systems and especially domains)

**Comment by stefanalund** [comment]
@freben times up, merging this now 😬

**Comment by freben** [comment]
Oh sorry, yes, this is fine!

**Comment by freben** [comment]
And agreed - I do think we should be opinionated on grouping concepts, especially along bounding context axes.

**Pull request merged**